File: queuesystem-core/src/main/java/com/example/queuesystemcore/ddd/kiosk/ui/rest/response/KioskConfiguration.java
package com.example.queuesystemcore.ddd.kiosk.ui.rest.response;

import com.example.queuesystemcore.ddd.kiosk.domain.KioskButton;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class KioskConfiguration {

    private String institutionName;
    private String city;
    private String street;
    private String houseNumber;
    private String postCode;
    private String postOffice;
    private String institutionLogoFile;
    private String backgroundColor;
    private List<KioskButton> kioskButtons;
}

File: kioskapp/src/dto/KioskConfiguration.ts

export type KioskConfiguration = {
    institutionName: string;
    city: string;
    street: string;
    houseNumber: string;
    postCode: string;
    postOffice: string;
    institutionLogoFile: string;
    backgroundColor: string;
    kioskButtons: KioskButton[];
}

export type KioskButton = {
    kioskButtonId: number;
    kioskSubButtonId: number;
    buttonText: string;
    queueConfigurationUUID: string;
}